pyro/distributions/diag_normal_mixture.py

```diff
@@ -67,7 +67,22 @@ def __init__(self, locs, coord_scale, component_logits):
         self.dim = locs.size(-1)
         self.categorical = Categorical(logits=component_logits)
         self.probs = self.categorical.probs
-        super(MixtureOfDiagNormals, self).__init__(batch_shape=batch_shape, event_shape=(self.dim,))
+        super(MixtureOfDiagNormals, self).__init__(batch_shape=torch.Size(batch_shape),
+                                                   event_shape=torch.Size((self.dim,)))
+
+    def expand(self, batch_shape, _instance=None):
```

Comment:
@neerajprad why are these expand methods necessary?

Reply:
We don't have `ReshapedDistribution` anymore, and we have implemented a `.expand` method for each of the distributions (in PyTorch). This is used when we call `dist.expand()`. If we are only overriding any existing distribution without changing its `__init__` method or using TransformedDistribution, we don't need to define it though.

Reply:
This was the original issue - https://github.com/uber/pyro/issues/1337. And this was the pytorch PR - https://github.com/pytorch/pytorch/pull/11341#pullrequestreview-153078931.